jujugui/static/gui/src/app/app.js

```diff
@@ -1226,6 +1226,45 @@ YUI.add('juju-gui', function(Y) {
     },
 
     /**
+      Renders the zoom component to the page in the designated element.
+
+      @method _renderZoom
+    */
+    _renderZoom: function() {
+      ReactDOM.render(
+        <window.juju.components.Zoom
+        zoomInCanvas={this._zoomInCanvas.bind(this)}
+        zoomOutCanvas={this._zoomOutCanvas.bind(this)} />,
+        document.getElementById('zoom-container'));
+    },
+
+    /**
+      Fires the zoom function with an increased zoom value
+
+      @method _zoomInCanvas
+    */
+    _zoomInCanvas: function() {
```

Comment:
this method and the one below don't appear to have any use in app.js, only in the component. Can you move them into the component and pass in `topo` to the component.